homeassistant/components/vesync/__init__.py

```diff
@@ -126,6 +134,16 @@ async def async_new_device_discovery(service):
             fans.extend(new_fans)
             hass.async_create_task(forward_setup(config_entry, "fan"))
 
+        light_set = set(light_devs)
+        new_lights = list(light_set.difference(lights))
+        if new_lights and lights:
+            lights.extend(new_lights)
+            async_dispatcher_send(hass, VS_DISCOVERY.format(VS_LIGHTS), new_lights)
+            return
+        if new_lights and not lights:
+            lights.extend(new_lights)
+            hass.async_create_task(forward_setup(config_entry, "light"))
```

Comment:
`switches`, `fan`, and `lights` all look like similar code.  We should make this more DRY in a future PR.